client/connection.cpp

```diff
@@ -103,6 +108,28 @@ Connection::Connection() noexcept
 		// TODO: Check errno. Should probably attach it to SocketError.
 		std::cerr << "RemoteCL Client failed to initialise." << std::endl;
 	}
+
+	mEventCallbackWorker = std::thread([&]() {
+		while (mActive) {
+			FireEventCallback E;
+
+			try {
+				mEventStream->read(E);
+				for (const auto &cb : mEventCallbacks) {
+					if (cb.ID == E.mData) {
+						cb.callback(cb.event, cb.callbackType, cb.userData);
+					}
+				}
+			} catch (Socket::Error) {
+				// PacketStream will throw this error when a terminate packet is received.
+				// The destructor will shut down the packet stream (and thus the associated socket
+				// stream as well), which throws a socket error in that case.
+				break;
+			} catch (...) {
+				// just keep on trying
```

Comment:
Do we know what situations could trigger other exceptions?
Does it make sense to continue under those situations?